### ISSUE TITLE
Use a fork of postcss-scss as dependency?

### ISSUE THREAD

**Issue by dryoma** [kristerkari/stylelint-scss#74]
The situation so far: PostCSS (with or without postcss-scss plugin) incorrectly parses [Sass nested properties](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#nested_properties). Right now `margin: 0 { left:10px; }` becomes
```js
{ type: 'rule', selector: 'margin: 0', ..., nodes: [ ... ] }
```
So to tell such nested groups from real rules we have to apply additional checks, i.e. [hacks like this](https://github.com/kristerkari/stylelint-scss/blob/nested/src/utils/parseNestedPropRoot.js).

In my opition (and [not only my](https://github.com/jedmao/postcss-nested-props/issues/10#issuecomment-243049285)) those should be parsed as declarations. A PR to postcss-scss with this implementation made it even worse. So I'd like to discuss creatinga **a fork to postcss-scss** with this behavior.

------------------

**The hack**
Pros:
+ We have full control over it and usually don't need to keep up with some other code.

Cons:
- Has to be applied in every instance where nested props could come up.
- Additional level of parsing stuff that is already parsed by PostCSS.
- (kinda questionable) If PostCSS does change, the hack will have to be updated.

**A fork for postcss-scss**
Pros:
+ Will make nested props parsed by postcss with no need in further complex checks (only basic ones, e.g. `if (item.isNested) {`.
+ Less processing work for cases like [this](https://github.com/kristerkari/stylelint-scss/blob/8ba304c2d1e0749a4625a19e8966dc3dc40f7ef1/src/rules/percent-placeholder-pattern/index.js#L31)

Cons:
- Will have to be updated every time the original project gets updated.
- On the other hand, in cases like [this](https://github.com/kristerkari/stylelint-scss/blob/366977a6367d0cfb5ce949c18d564689d60a029d/src/rules/dollar-variable-no-missing-interpolation/index.js#L82) and [here](https://github.com/kristerkari/stylelint-scss/blob/366977a6367d0cfb5ce949c18d564689d60a029d/src/rules/dollar-variable-no-missing-interpolation/index.js#L66) additional checks might get necessary.

---

Additional moments:

```scss
$var: margin;

a {
  #{$var}: 10px { // Here a variable _has_ to have interpolation, otherwise it's an error
    origin: left;
  }
}
```

**Comment by dryoma** [comment]
Cases in our code that will (presumably) benefit from a **fork** (i.e. additional checks won't be required):
* `dollar-variable-no-missing interpolation`:  [here](https://github.com/kristerkari/stylelint-scss/blob/366977a6367d0cfb5ce949c18d564689d60a029d/src/rules/dollar-variable-no-missing-interpolation/index.js#L120) nested groups namespace prop value (`margin: $var { ... }`) is missed
* `operator-no-unspaces` is skipping root nested decls at the moment ([here](https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/operator-no-unspaced/index.js#L137) and [here](https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/operator-no-unspaced/index.js#L150))
* and of course [`declaration-nested-properties-` rules](https://github.com/kristerkari/stylelint-scss/pull/63)

Cases that **will require** additional checks with a **fork**:
* `dollar-variable-no-missing interpolation`:  [here](https://github.com/kristerkari/stylelint-scss/blob/366977a6367d0cfb5ce949c18d564689d60a029d/src/rules/dollar-variable-no-missing-interpolation/index.js#L82) nested props will be ignored since `margin: 0 { ... }` will not be a Rule anymore.

**Comment by evilebottnawi** [comment]
@dryoma maybe create issue in `postcss-scss`, it would be better then create fork?

**Comment by dryoma** [comment]
@evilebottnawi I have already done so, and even made a PR. Long story short - it became even worse.

**Comment by evilebottnawi** [comment]
@dryoma can you give the link to issue? We definitely need to solve the problem correctly :+1:

**Comment by dryoma** [comment]
The [discussion](https://github.com/postcss/postcss-scss/issues/39), the [PR](https://github.com/postcss/postcss-scss/pull/40).

**Comment by evilebottnawi** [comment]
@dryoma hm
`a { margin: 0 { left:10px; } }`
For `margin`:
```shell
NestedDeclaration {
  raws: 
   { before: ' ',
     between: ': ',
     value: { value: '0', raw: '0 ' },
     semicolon: true,
     after: ' ' },
  type: 'decl',
  isNested: true,
  nodes: 
   [ Declaration {
       raws: [Object],
       type: 'decl',
       parent: [Circular],
       source: [Object],
       prop: 'left',
       value: '10px' } ],
  parent: 
   Rule {
     raws: { before: '', between: ' ', semicolon: false, after: ' ' },
     type: 'rule',
     nodes: [ [Circular] ],
     parent: 
      Root {
        raws: [Object],
        type: 'root',
        nodes: [Object],
        source: [Object],
        lastEach: 4,
        indexes: {} },
     source: { start: [Object], input: [Object], end: [Object] },
     selector: 'a',
     lastEach: 4,
     indexes: {} },
  source: 
   { start: { line: 1, column: 5 },
     input: Input { css: 'a { margin: 0 { left:10px; } }', id: '<input css 1>' },
     end: { line: 1, column: 28 } },
  prop: 'margin',
  value: '0',
  lastEach: 4,
  indexes: {} }
```
i see `isNested`, what about create util or repo `resolveNestedProperties` and put logic there. 
I misunderstood what you wish to receive from `postcss-scss`. About interpolation in `Additional` - i think it is bug and should be fixed in `postcss-scss`. In my opinion it has nothing related with `nested properties`.

**Comment by dryoma** [comment]
@evilebottnawi I describe the problem in [this comment](https://github.com/postcss/postcss-scss/issues/39#issuecomment-243105041): `a { margin: 0 { left:10px; } }` is a Declaration, `a { margin: { left:10px; } }` is a Rule (seems I mistyped it in the first message).

> what about create util or repo resolveNestedProperties
There is already [one](https://github.com/kristerkari/stylelint-scss/blob/master/src/utils/parseNestedPropRoot.js), the question is whether we really should leave it as is.
